tests/analytical.py

```diff
@@ -1,88 +1,113 @@
 import numpy as np
 
 
-def gradient(x_a, y_a, z_a, x_b, y_b, z_b, c_a, c_b):
+def gradient(coordinates,charges):
     """Computes the gradient."""
+    number_coordinates = len(coordinates)
+    eval_E = []
+    for atom_a in range(number_coordinates):
+        for atom_b in range(number_coordinates):
+            if atom_a < atom_b:
+                cxc = charges[atom_a] * charges[atom_b]
+                xyz_squared = ((coordinates[atom_a,0] - coordinates[atom_b,0]) ** 2 + (coordinates[atom_a,1] - coordinates[atom_b,1]) ** 2 + (coordinates[atom_a,2] - coordinates[atom_b,2]) ** 2) ** (3 / 2)
```

Comment:
You can get this a bit more easily and concisely with:
```python
xyz_32 = np.linalg.norm(coordinates[atom_a, :] - coordinates[atom_b, :])**3
```
Step by step:
1. `coordinates[atom_a, :] - coordinates[atom_b, :]` takes the difference between the rows with indices `atom_a` and `atom_b`. Numpy indexing notation documentation: https://numpy.org/doc/stable/user/basics.indexing.html
2. The `norm` function calculates the norm of a vector.
3. We take the third power.